stl/inc/mdspan

```diff
@@ -25,9 +25,19 @@ _STL_DISABLE_CLANG_WARNINGS
 #undef new
 
 _STD_BEGIN
+template <class _IndexType, size_t _Size>
+struct _Maybe_empty_array {
+    array<_IndexType, _Size> _Array{};
+};
+
+template <class _IndexType>
+struct _Maybe_empty_array<_IndexType, 0> {};
+
+template <size_t... _Extents>
+inline constexpr size_t _Calculate_rank_dynamic = ((_Extents == dynamic_extent) + ... + 0);

```

Comment:
Suggestion: This is adding `bool`s which is kind of weird. On left 43 / right 53 there used to be a `static_cast` making this clearer. I think a `static_cast<size_t>` could help make things clearer here (a rare case where adding unnecessary code might be a good idea).